NotificationHubs/test/com/windowsazure/messaging/e2e/InstallationCrudsE2E.java

```diff
@@ -91,12 +99,23 @@ public void BasicCrudScenarioTest() throws Exception{
 		assertNotNull(installation.getTemplates());
 		assertEquals(1, installation.getTemplates().size());
 		assertTrue(installation.getTemplates().get("template1").getBody().equalsIgnoreCase("{\"data\":{\"key2\":\"value2\"}}"));
-		assertNull(installation.getTemplates().get("template1").getTags());
+		templateTags = installation.getTemplates().get("template1").getTags();
+		assertNotNull(templateTags);
+		assertEquals(1, templateTags.size());
+		assertTrue(templateTags.get(0).equalsIgnoreCase("template1"));
 		assertNull(installation.getSecondaryTiles());
 		
+		PartialUpdateOperation removeUserId = new PartialUpdateOperation(UpdateOperationType.Remove, "/userId");
+		hub.patchInstallation(installation.getInstallationId(), removeUserId);
+		Thread.sleep(3000);
```

Comment:
Because of the change you made to `patchInstallation` above, I don't think you should need to `Thread.sleep` here.

Reply:
Actually no. patchInstallation return result immediately and push request in queue. So to get installation with changes we have to Thread.sleep for some time